### PR TITLE
BL-2169: Stop leaking image files

### PR DESCRIPTION
**Pull request by phillip-hopper**
This was leaking thousands of image files in Bloom.

### CODE REVIEW

File: PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs [hunk 1]
```diff
@@ -348,6 +351,11 @@ protected virtual void Dispose(bool disposing)
 					Image.Dispose();
 					Image = null;
 				}
+
+				if (!string.IsNullOrEmpty(LeakedFilePath))
+				{
+					File.Delete(LeakedFilePath);
```
Comment (hatton) [inline PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs]: Perhaps swallow any exceptions on this when not in #Debug?

File: PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs [hunk 2]
```diff
@@ -238,20 +238,23 @@ private static Image LoadImageWithoutLocking(string path)
...
 			var i = new PalasoImage
 			{
-				Image = LoadImageWithoutLocking(path),
+				Image = LoadImageWithoutLocking(path, out tempPath),
 				FileName = Path.GetFileName(path),
 				_originalFilePath = path,
 				_pathForSavingMetadataChanges = path,
-				Metadata = Metadata.FromFile(path)
+				Metadata = Metadata.FromFile(path),
+				LeakedFilePath = tempPath
```
Comment (hatton) [inline PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs]: It's not going to "Leaked" anymore, right? So how about "_tempFilePath"

File: PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs [hunk 3]
```diff
@@ -7,15 +7,15 @@
...
-using System.Linq;
 
 namespace Palaso.UI.WindowsForms.ImageToolbox
 {
 	public class PalasoImage : IDisposable
-
 	{
 		private Metadata _metadata;
 
+		private string LeakedFilePath { get; set; }
```
Comment (hatton) [inline PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs]: Any reason to be a property instead of a member?

Comment (jasonleenaylor) [inline PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs]: I'm not a fan of the variable name. It does follow from the local variable that you are exposing through the out parameter, but could you name it TemporaryFilePath or something else?

File: PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs (hunk 1 above)
Comment (phillip-hopper) [inline PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs]: Done.

File: PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs (hunk 3 above)
Comment (phillip-hopper) [inline PalasoUIWindowsForms/ImageToolbox/PalasoImage.cs]: Done.

**Comment by hatton** [comment]
:+1:

**Pull request merged**